Use pkg_resources for version detection

tehsmyers opened the pull request:
pbr's get_version is apparently not useful except when making a python
package, which (in hindsight) makes perfect sense. Use pkg_resources
instead to get information about the currently installed version of
miqcli.

Also fixed a similar block in the doc conf.py

fixes #60

tehsmyers commented:
This is the simplest approach to solving the problem, but I mentioned some other potential solutions in #60

tvieira commented:
Wouldn't `except Exception` capture pbr exception if it happens? Because then just adding the base type `Exception` would solve the E722 error.

tehsmyers commented:
Yeah, I'm not sure why my local linter was just fine with that. Regardless, I'll get it fixed up.

tehsmyers commented:
So that's fixed, but https://github.com/rhpit/manageiq-cli/issues/60#issuecomment-358428752 is still a problem, since that implies --version will never work even in a "real" install of miqcli.

ryankwilliams commented on miqcli/constants.py:
```diff
@@ -25,8 +25,15 @@
 #: name of miqcli package
 PACKAGE = 'miqcli'
 
-#: Installed version of miqcli
```
I am still unsure about using pbr `get_version` function within the constants module to set the version. Each time the CLI is run it will use pbr to get the version information. Pbr's function tries 3 different locations to get this information.

1. By an environment variable
2. From the package metadata (distribution)
3. From git

I created a test package on the PyPI test server with a very simple Python package setup using pbr. Once I installed the package and tried to run. I was unable to proceed, I was getting the same exception mentioned in #60. My opinion is when this is live and ready on PyPI, we will not be able to get the version from pbr. It is going to have to be defined within the constants module.

tehsmyers replied:
The version should never be statically defined in the constants module, lest we be forced to update it constantly. One of the biggest reasons to use pbr is that your version is abstracted and you never again have to care about it except for the purposes of tagging releases in git. My usage in this case is just an abuse of pbr, because (as the name would imply despite my ignorance) pbr is for packaging, not...not packaging(?).

The version should always be detectable from the environment, and this appears to work reliably when going through `pkg_resources` to do it. I just gave up too easily when it didn't work once for me, and I never noticed that I was using *the wrong python interpreter*.

ryankwilliams replied:
I should have mentioned in my comment that this is first time working with pbr. You are right, having the capability provided by pbr to handle the versioning is great!

tehsmyers replied:
I've been using pbr pretty happily for years, and frankly should have known better when I encountered that error with the "setuptoolsy" `get_distribution` rather than go searching for some other way to do it. I'm very glad you caught this quickly.

tehsmyers commented:
pr updated to abandon pbr's `get_version`

Commit:
Use pkg_resources for version detection

pbr's get_version is apparently not useful except when making a python
package, which (in hindsight) makes perfect sense. Use pkg_resources
instead to get information about the currently installed version of
miqcli.

Also fixed a similar block in the doc conf.py

ryankwilliams reviewed (approved):
LGTM

tehsmyers commented:
```sh
(miqcli) @ $ pip install -e manageiq-cli/             
...
Successfully installed miqcli

$ cd manageiq-cli/

$ miqcli --version                                                                                                             
Installed version : 0.0.1.dev24                                                                                            
Latest version    : N/A                                                                                                            
                                                                                                                                           
You are running a pre-release version of ManageIQ CLI!

$ cd ..                                                                                                           

(miqcli) @ $ miqcli --version
Installed version : 0.0.1.dev24
Latest version    : N/A        
                       
You are running a pre-release version of ManageIQ CLI!

(miqcli) @ $ pip install -U manageiq-cli/
...
Successfully installed miqcli-0.0.1.dev24

$ cd manageiq-cli/

(miqcli) @ $ miqcli --version
Installed version : 0.0.1.dev24
Latest version    : N/A

You are running a pre-release version of ManageIQ CLI!

(miqcli) @ $ cd ..

$ miqcli --version
Installed version : 0.0.1.dev24
Latest version    : N/A

You are running a pre-release version of ManageIQ CLI!
```

tvieira reviewed (approved):
LGTM
